server/game/gameSystems/DiscardCardsFromHand.ts

```diff
@@ -0,0 +1,84 @@
+import { EventName, Location, RelativePlayer, TargetMode } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import { Card } from '../core/card/Card';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+
+export interface IDiscardCardsFromHandProperties extends IPlayerTargetSystemProperties {
+    amount?: number;
+    cardCondition?: (card: Card, context: AbilityContext) => boolean;
+}
+
+export class DiscardCardsFromHand<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardCardsFromHandProperties> {
+    protected override defaultProperties: IDiscardCardsFromHandProperties = {
+        amount: 1,
+        cardCondition: () => true
+    };
+
+    public override name = 'discard';
+    public override eventName = EventName.OnCardsDiscardedFromHand;
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public override eventHandler(_event): void { }
+
+    public override getEffectMessage(context: TContext): [string, any[]] {
+        const properties = this.generatePropertiesFromContext(context);
+        return ['make {0} discard {1} cards', [properties.target, properties.amount]];
+    }
+
+    public override canAffect(playerOrPlayers: Player | Player[], context: TContext, additionalProperties = {}): boolean {
+        const players = Array.isArray(playerOrPlayers) ? playerOrPlayers : [playerOrPlayers];
```

Comment:
Just a heads-up if you find it useful, I also have `asArray` in Helper.ts for this situation. So you could directly do `return Helpers.asArray(players).every(...)`